### ISSUE TITLE
Iterate over metric handling and providing in Zeebe

### ISSUE THREAD

**Issue by ChrisKujawa** [camunda/camunda#11570]
**Description**

Right now we have a lot of metrics in our system, to me, it is not 100% clear what the impact is on performance (but I doubt it is for free). 

Ideally, I would like to have something similar to what we could have with our logging, where you can disable logging for certain packages and the same for metrics. 

This means I would like to have multiple configuration (maybe for each module :thinking: ) where I can disable and enable metrics. In that way we can continue adding more metrics even fine-granular metrics but not enabling them by default, e.g. actor metrics or column family metrics etc. 

This would also allow us to measure the impact of the metrics on our performance (we can test then how we perform with less metric or with all etc.)

**Comment by npepinpe** [comment]
Yes, a metrics facade would be amazing. Either Micrometer or OpenTelemetry, whichever is more flexible :+1: :100: :fire:

**Comment by npepinpe** [comment]
The crux here is finding the right arguments in order to prioritize this.

- This would make it much easier to test metrics, since we rely heavily on some of them.
- This could reduce costs and improve performance by allowing us to disable metrics at varying level of granularity

Anything else?

**Comment by Zelldon** [comment]
Would using OpenTelemetry for this also help us in making progress towards tracing maybe?

**Comment by npepinpe** [comment]
A little, since you'd already include the API and set up the distribution module for it. So it will simplify things a bit, for sure.

**Comment by oleschoenburg** [comment]
Linking #11267 as it is at least somewhat related.

**Comment by megglos** [comment]
Sidenote: 
- we experienced performance drop using a prototype that contained additional metrics @megglos link it here
- there is an interest of users to have flags to enable disable metrics to do root-causing of performance issues - but disable them for production once the system setup is optimized

**Comment by megglos** [comment]
ZDP-Planning:
- worth a refactoring and would give us more control + allow assessment of performance impact
- we didn't get to it since March, moving to backlog for later

**Comment by npepinpe** [comment]
Also, FYI, Prometheus just merged native support for OTLP, so at some point we might be able to just drop Prometheus entirely and push out OTLP only, letting users configure the actual endpoint.

**Comment by Zelldon** [comment]
Planning:

* We started to introduce micrometer, we are planning to migrate more metrics to it (as a facade)
* Will not happen before 8.3
